### PR TITLE
Bttl 800c globals

### PR DESCRIPTION
**Pull request by theflyingzamboni**
Typed, named, and refactored as many of the globals in 800c as I could, along with quite a bit of code that uses them. Anything I refactored into a struct should probably be double-checked in-game to make sure it still works correctly. I've caught some bugs I accidentally introduced, but it's possible I missed a couple.

### CODE REVIEW

File: src/main/java/legend/game/Scus94491BpeSegment.java [hunk 1]
```diff
@@ -1412,15 +1412,14 @@ public static void FUN_80018a5c(final int x, final int y, final int leftU, final
...
+  public static void renderButtonPressHudTexturedRect(final int x, final int y, final int u, final int v, final int width, final int height, final int clutOffset, @Nullable final Translucency transMode, final COLOUR colour, final int a9) {
+    renderButtonPressHudElement((short)x, (short)y, u & 0xff, v & 0xff, u + width & 0xff, v + height & 0xff, clutOffset, transMode, colour, a9, a9);
   }
 
   @Method(0x80018decL)
-  public static void FUN_80018dec(final int x, final int y, final int u, final int v, final int width, final int height, final int clutButNotExactly, @Nullable final Translucency transMode, final COLOUR colour, final long a9, final long a10) {
-    FUN_80018a5c(x, y, u, v, u + width, v + height, clutButNotExactly, transMode, colour, a9, a10);
+  public static void renderDivineDragoonAdditionPressIris(final int x, final int y, final int u, final int v, final int width, final int height, final int clutOffset, @Nullable final Translucency transMode, final COLOUR colour, final int a9, final int a10) {
```
Comment (Ink230) [inline src/main/java/legend/game/Scus94491BpeSegment.java:1421]: long to int?

File: src/main/java/legend/game/combat/Bttl_800f.java [hunk 2]
```diff
@@ -130,64 +131,64 @@ private Bttl_800f() { }
...
       final GpuCommandPoly cmd = new GpuCommandPoly(4)
         .bpp(Bpp.BITS_4)
         .clut(720, 497)
         .vramPos(704, 256)
         .monochrome(0x80)
-        .pos(0, left, top)
-        .pos(1, right, top)
-        .pos(2, left, bottom)
-        .pos(3, right, bottom)
+        .pos(0, leftX, topY)
```
Comment (Ink230) [inline src/main/java/legend/game/combat/Bttl_800f.java:188]: Does appending X and Y to the left/right, top/bottom directions make sense in this context? If there can't be a leftY why does it need to be noted there is a leftX?

**Review by Ink230 — COMMENTED** [review]
Some comments just for confirming sanity.

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/game/Scus94491BpeSegment_8003.java [hunk 3]
```diff
@@ -1951,15 +1951,15 @@ public static int perspectiveTransformTriple(final SVECTOR world0, final SVECTOR
...
 
   @Method(0x8003f990L)
-  public static void RotTrans(final SVECTOR v0, final VECTOR out, @Nullable final UnsignedIntRef flags) {
-    CPU.MTC2(v0.getXY(), 0); // VXY0
-    CPU.MTC2(v0.getZ(), 1); // VZ0
-    CPU.COP2(0x48_0012L); // MVMVA (translation=tr, mul vec=v0, mul mat=rot, 12-bit fraction)
+  public static void RotTrans(final SVECTOR in, final VECTOR out, @Nullable final UnsignedIntRef flags) {
+    CPU.MTC2(in.getXY(), 0); // VXY0
+    CPU.MTC2(in.getZ(), 1); // VZ0
+    CPU.COP2(0x48_0012L); // MVMVA (translation=tr, mul vec=in, mul mat=rot, 12-bit fraction)
```
[the remark this answers is not in the document]
Comment (theflyingzamboni) [reply src/main/java/legend/game/Scus94491BpeSegment_8003.java:1962]: loooool I thought that was just another generic register variable name

File: src/main/java/legend/game/Scus94491BpeSegment.java (hunk 1 above)
Comment (theflyingzamboni) [reply src/main/java/legend/game/Scus94491BpeSegment.java:1421]: They aren't addresses or int flags, so they should never over flow. From what I can see, they'd never even overflow a short.

File: src/main/java/legend/game/combat/Bttl_800f.java (hunk 2 above)
Comment (theflyingzamboni) [reply src/main/java/legend/game/combat/Bttl_800f.java:188]: They're distinguishing e.g. topY from topV or leftX from leftU.

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**